src/features/dotnetTest.ts

```diff
@@ -13,236 +12,284 @@ import * as utils from '../common';
 import * as net from 'net';
 import * as os from 'os';
 import * as path from 'path';
+import TelemetryReporter from 'vscode-extension-telemetry';
+import AbstractProvider from './abstractProvider';
 
-let _testOutputChannel: vscode.OutputChannel = undefined;
+const TelemetryReportingDelay = 2 * 60 * 1000; // two minutes
 
-function getTestOutputChannel(): vscode.OutputChannel {
-    if (_testOutputChannel === undefined) {
-        _testOutputChannel = vscode.window.createOutputChannel(".NET Test Log");
+export default class TestManager extends AbstractProvider {
+    private _channel: vscode.OutputChannel;
+
+    private _runCounts: { [testFrameworkName: string]: number };
+    private _debugCounts: { [testFrameworkName: string]: number };
+    private _telemetryIntervalId: NodeJS.Timer = undefined;
+
+    constructor(server: OmniSharpServer, reporter: TelemetryReporter) {
+        super(server, reporter);
+
+        // register commands
+        let d1 = vscode.commands.registerCommand(
+            'dotnet.test.run',
+            (testMethod, fileName, testFrameworkName) => this._runDotnetTest(testMethod, fileName, testFrameworkName));
+
+        let d2 = vscode.commands.registerCommand(
+            'dotnet.test.debug',
+            (testMethod, fileName, testFrameworkName) => this._debugDotnetTest(testMethod, fileName, testFrameworkName));
+
+        this._telemetryIntervalId = setInterval(() =>
```

Comment:
Potentially, though I don't think it's huge deal.

I think what I really want to do here is have a shared component that sends telemetry like this. This is *really* similar to the telemetry reporting in the OmniSharpServer class, and I'd like to unify them. Are you OK if I leave this for now and unify later? I've filed #1693 to track unifying these in either case.